tests/cpp/test_scatter.cpp

```diff
@@ -168,7 +168,69 @@ TEST_F(ScatterTest, GridCounting) {
   testValidate(&fusion, outputs, {t0}, __LINE__, __FILE__);
 }
 
-TEST_F(ScatterTest, BlockCountingWithShmem2D) {
+TEST_P(ScatterTest, BlockCountingWithShmem2DExact) {
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
+
+  const std::vector<int64_t> self_shape{4, 100};
+  const std::vector<int64_t> index_shape{4, 10};
+
+  auto tv0 = makeContigConcreteTensor(index_shape, DataType::Int);
+  fusion.addInput(tv0);
+
+  auto tv1 = set(tv0);
+  auto tv2 = zeros(
+      {IrBuilder::create<Val>(self_shape[0]),
+       IrBuilder::create<Val>(self_shape[1])},
+      DataType::Int);
+  auto tv3 = ones(
```

Comment:
I think we can use a non-one factory method to properly test this. i.e. should we take this as an input instead?

Reply:
I'm not sure why not using ones would increase the test coverage.